auth/options.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2025 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"net/url"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/fluxcd/pkg/cache"
+)
+
+// Option is a functional option for the auth package.
+type Option func(*Options)
+
+// Options contains options for configuring the behavior of the provider methods.
+// Not all providers/methods support all options.
+type Options struct {
+	Client          client.Client
+	Cache           *cache.TokenCache
+	ServiceAccount  *client.ObjectKey
+	InvolvedObject  cache.InvolvedObject
+	Scopes          []string
+	ImageRepository string
```

Comment:
I suggest using `Artifact` instead of `Image` throughout the APIs:

- `ArtifactRepository`
- `WithArtifactRepository`
- `GeArtifactCacheKey`
- `ArtifactRegistry` 
- `NewArtifactRegistryToken`
- `ArtifactRegistryCredentials`

Reply:
Fixed, thanks!

Reply:
Oops, missed a few, fixing